src/expr/src/scalar/like_pattern.rs

```diff
@@ -16,6 +17,53 @@ use serde::{Deserialize, Serialize};
 use crate::scalar::EvalError;
 use mz_lowertest::MzReflect;
 
+/// The escape string to use by default in LIKE patterns.
+pub const DEFAULT_ESCAPE: &str = "\\";
+
+/// Converts a pattern string that uses a custom escape character to one that uses the default.
+fn normalize_pattern(pattern: &str, escape: &str) -> Result<String, EvalError> {
+    if escape.eq(DEFAULT_ESCAPE) {
+        return Ok(String::from(pattern));
+    }
+    let default_escape_char: char = DEFAULT_ESCAPE.chars().next().unwrap();
+    let mut p = String::with_capacity(2 * pattern.len());
+    if escape.is_empty() {
+        for c in pattern.chars() {
+            if c == default_escape_char {
+                p.push(c);
+            }
+            p.push(c);
+        }
+    } else {
+        let mut ecs = escape.chars();
+        let custom_escape_char: char = ecs.next().unwrap();
+        if !ecs.next().is_none() {
+            return Err(EvalError::LikeEscapeTooLong);
+        }
+        let mut cs = pattern.chars();
+        loop {
```

Comment:
rust nit: this might be more idiomatic? https://doc.rust-lang.org/rust-by-example/flow_control/while_let.html
`while let Some(c) = cs.next() {`

Reply:
Much nicer... thanks!